chore: optimize bitflags in NetworkTransformState

EmandM opened the pull request:
## Purpose of this PR

Remove the `GetFlag` and `SetFlag` functions from `NetworkTransformState`

### Jira ticket

[MTT-13655](https://jira.unity3d.com/browse/MTT-13655)
[MTTB-1763](https://jira.unity3d.com/browse/MTTB-1763)

fix: #3788

### Changelog

- Changed: Optimized performance of `NetworkTransformState

## Documentation

- No documentation changes or additions were necessary.

## Testing & QA (How your changes can be verified during release Playtest)
[//]: #  (
This section is REQUIRED and should describe how the changes were tested and how should they be tested when Playtesting for the release.
It can range from "edge case covered by unit tests" to "manual testing required and new sample was added".
Expectation is that PR creator does some manual testing and provides a summary of it here.)

<!-- Add any performance testing results here if relevant. -->

### Functional Testing
[//]: # (If checked, List manual tests that have been performed.)
_Manual testing :_
- [ ] `Manual testing done`

_Automated tests:_
- [ ] `Covered by existing automated tests`
- [ ] `Covered by new automated tests`

_Does the change require QA team to:_

- [ ] `Review automated tests`?
- [ ] `Execute manual tests`?
- [ ] `Provide feedback about the PR`?

If any boxes above are checked the QA team will be automatically added as a PR reviewer.

## Backports

No backport needed.

EmandM committed:
chore: optimize bitflags in NetworkTransformState

EmandM committed:
Update CHANGELOG

NoelStephensUnity committed:
Update IgnoreIfServiceEnvironmentVariableSetAttribute.cs

NoelStephensUnity commented on com.unity.netcode.gameobjects/Runtime/Components/NetworkTransform.cs:
```diff
@@ -997,6 +844,72 @@ public void NetworkSerialize<T>(BufferSerializer<T> serializer) where T : IReade
             }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public void SerializeBitset(ref FastBufferWriter writer)
```
```suggestion
            internal void SerializeBitset(ref FastBufferWriter writer)
```

NoelStephensUnity commented on com.unity.netcode.gameobjects/Runtime/Components/NetworkTransform.cs:
```diff
@@ -997,6 +844,72 @@ public void NetworkSerialize<T>(BufferSerializer<T> serializer) where T : IReade
+            }
+
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public void DeserializeBitset(ref FastBufferReader reader)
```
```suggestion
            internal void DeserializeBitset(ref FastBufferReader reader)
```

NoelStephensUnity reviewed (commented):
Still reviewing but have two places that I think you meant to make those methods internal and not public API.

NoelStephensUnity commented on com.unity.netcode.gameobjects/Tests/Runtime/TestHelpers/NUnitExtensions/IgnoreIfServiceEnvironmentVariableSetAttribute.cs:
```diff
@@ -0,0 +1,26 @@
+namespace Unity.Netcode.TestHelpers.Runtime
+{
+    [AttributeUsage(AttributeTargets.Class, AllowMultiple = false, Inherited = true)]
+    public class IgnoreIfServiceEnvironmentVariableSetAttribute : NUnitAttribute, IApplyToTest
```
```suggestion
    internal class IgnoreIfServiceEnvironmentVariableSetAttribute : NUnitAttribute, IApplyToTest
```

EmandM committed:
Apply suggestions from code review

Co-authored-by: Noel Stephens <noel.stephens@unity3d.com>

NoelStephensUnity committed:
style

removing and adding whitespaces.

NoelStephensUnity committed:
Merge branch 'develop-2.0.0' into chore/optimize-network-transform-state

NoelStephensUnity committed:
update

Merge fix.

EmandM committed:
Fix cumulative state

EmandM committed:
Merge branch 'develop-2.0.0' into chore/optimize-network-transform-state

netcode-ci-service committed:
Merge develop-2.0.0 into chore/optimize-network-transform-state

netcode-ci-service committed:
Merge develop-2.0.0 into chore/optimize-network-transform-state

netcode-ci-service committed:
Merge develop-2.0.0 into chore/optimize-network-transform-state

netcode-ci-service committed:
Merge develop-2.0.0 into chore/optimize-network-transform-state

netcode-ci-service committed:
Merge develop-2.0.0 into chore/optimize-network-transform-state

netcode-ci-service committed:
Merge develop-2.0.0 into chore/optimize-network-transform-state

netcode-ci-service committed:
Merge develop-2.0.0 into chore/optimize-network-transform-state

netcode-ci-service committed:
Merge develop-2.0.0 into chore/optimize-network-transform-state

netcode-ci-service committed:
Merge develop-2.0.0 into chore/optimize-network-transform-state

EmandM committed:
Fix whitespace issue

NoelStephensUnity committed:
fix

Fixing AnticipatedNetworkTransform m_NetworkManager issue.

NoelStephensUnity committed:
refactor

Refactoring how we handle updating public facing flags.

NoelStephensUnity committed:
update

adding comment

NoelStephensUnity committed:
update

Updating some tests with these changes
(WIP) Migrating away from direct transform access and using a cached transform approach.

NoelStephensUnity committed:
refactor

Migrating all of the flags into FlagStates.
Migrating the bit serialization into FlagStates.
Migrating flag helper methods into FlagStates.

NoelStephensUnity committed:
fix

This resolves the issue with the EntityId no longer supporting integer conversion.

NoelStephensUnity committed:
style

removing partial comment and whitespace.

NoelStephensUnity committed:
fix

Another spot where EntityId was needed.

NoelStephensUnity committed:
update

Missed SwitchTransformSpaceWhenParented.

NoelStephensUnity committed:
refactor - EntityId

Since MP Tools subscribes to the TransportInitialized and TransportDisposed methods and we have to vet against packages with dependencies, I am putting a temporary work-around to get past this issue so we can land the EntityId fix.
Once MP Tools is updated for EntityId, we can re-enable the original fix.

NoelStephensUnity committed:
update

Found a few more places that could be micro-optimized.
Fixing missed transformToUse that was if-def'd out.

EmandM committed:
Small naming changes

EmandM committed:
Cleanup files

EmandM committed:
Unbreak breaking change

NoelStephensUnity reviewed (approved):
👍
